### PR TITLE
Add option to provide API Group for customizable PKI

### PR DESCRIPTION
**Pull request by shubhamcoc**
## Description

Ability to add API group for the customizable PKI provided by the user. 
 
## Type of Change

- [x] New Feature (#911 )

## Checklist

- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)

### CODE REVIEW

**Comment by panyuenlau** [comment]
Quick question @shubhamcoc - have you tested it with cert-manager in an actual k8s cluster?

**Comment by shubhamcoc** [comment]
Hi @panyuenlau, @pregnor Do you guys mean testing in kind cluster or we need some kind of customized cert manager for testing?

**Comment by panyuenlau** [comment]
> Hi @panyuenlau, @pregnor Do you guys mean testing in kind cluster or we need some kind of customized cert manager for testing?

What I mean by testing is to:
1. Follow the [quick start](https://github.com/banzaicloud/koperator#quick-start) to bring up a running Kafka cluster, and you would need to build and use your customized image when you install the koeprator, for example:
```
helm install kafka-operator --repo https://kubernetes-charts.banzaicloud.com kafka-operator --namespace=kafka --create-namespace --set operator.image.repository=<path-to-your-repository>/<your-image-name> --set operator.image.tag=<your-image-tag>
```
2. You would also need to install cert manager to the k8s cluster, one way of doing that is 
```
kubectl apply \
--validate=false \
-f https://github.com/jetstack/cert-manager/releases/download/v1.11.0/cert-manager.crds.yaml
```
3. And lastly you need to create a `KafkaUser` with the `group` info to verify the behavior that you updated in the PR:
```
apiVersion: kafka.banzaicloud.io/v1alpha1
kind: KafkaUser
metadata:
  name: example-kafkauser-with-pki
  namespace: kafka
spec:
  clusterRef:
    name: kafka
  secretName: example-kafkauser-secret
  pkiBackendSpec:
    pkiBackend: "cert-manager"
    issuerRef:
      name: "ca-issuer"
      kind: "ClusterIssuer"
      group: <supported-group-by-cert-manager>
```
A `Certificate` CR should be created successfully by cert-manager after this

**Comment by panyuenlau** [comment]
I also verified with 
```
kubectl apply -f -<<EOF
apiVersion: kafka.banzaicloud.io/v1alpha1
kind: KafkaUser
metadata:
  name: example-kafkauser
  namespace: kafka
spec:
  clusterRef:
    name: kafka
  secretName: example-kafkauser-secret
  pkiBackendSpec:
    pkiBackend: "cert-manager"
    issuerRef:
      name: "kafka-kafka-issuer"
      kind: "ClusterIssuer"
      group: cert-manager.io
EOF
```

Where `kafka-kafka-issuer` being the cluster issuer:
```
kubectl get clusterissuer kafka-kafka-issuer -o yaml                                                                                                                     1 ↵
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  creationTimestamp: "2023-05-05T18:57:53Z"
  generation: 1
  labels:
    app: kafka
    kafka_issuer: kafka-kafka-issuer
  name: kafka-kafka-issuer
  resourceVersion: "41574"
  uid: 4335bbf3-0292-4519-bfc0-ad54eb5003e6
spec:
  ca:
    secretName: kafka-ca-certificate
status:
  conditions:
  - lastTransitionTime: "2023-05-05T19:18:58Z"
    message: Signing CA verified
    observedGeneration: 1
    reason: KeyPairVerified
    status: "True"
    type: Ready
```

**Review by panyuenlau — APPROVED** [review]
Thanks!

I also see that improvements of the relevant tests can be made in 
https://github.com/banzaicloud/koperator/blob/636484581697bc096b007ccb787e1b5b857d2104/pkg/pki/certmanagerpki/certmanager_user_test.go#L64

But I think I will open another PR for improving the existing tests once I get a chance

**Review by pregnor — APPROVED** [review]

File: pkg/pki/certmanagerpki/certmanager_user.go [hunk 1]
```diff
@@ -182,7 +183,7 @@ func (c *certManager) clusterCertificateForUser(
 }
 
 // getCA returns the CA name/kind for the KafkaCluster
```
Comment (bartam1) [inline pkg/pki/certmanagerpki/certmanager_user.go:185]: We should modify the description to name/kind/group

**Pull request merged**